source/javascripts/hooks/useEnvVars.ts

```diff
@@ -0,0 +1,159 @@
+import { useMemo } from 'react';
+import { useShallow } from 'zustand/react/shallow';
+import { useQueries, useQuery } from '@tanstack/react-query';
+import useBitriseYmlStore from '@/hooks/useBitriseYmlStore';
+import StepService from '@/core/models/StepService';
+import { EnvVar, EnvVarYml } from '@/core/models/EnvVar';
+import EnvVarService from '@/core/models/EnvVarService';
+import { WorkflowYmlObject } from '@/core/models/Workflow';
+import EnvVarsApi from '@/core/api/EnvVarsApi';
+import WindowUtils from '@/core/utils/WindowUtils';
+import StepApi from '@/core/api/StepApi';
+
+function envVarsAreEqual(a: EnvVar, b: EnvVar) {
+  return a.key === b.key && a.source === b.source;
+}
+
+const useCvsFromYml = (workflowId: string) => {
+  return useBitriseYmlStore(
+    useShallow((s) => {
+      const cvss: string[] = [];
+
+      const collectStepCvssOf = (workflow?: WorkflowYmlObject) => {
+        workflow?.steps?.forEach((ymlStepObject) => {
+          const [cvs, step] = Object.entries(ymlStepObject)[0];
+
+          // TODO: Handle step bundles and with groups...
+          if (!StepService.isStepBundle(cvs, step) && !StepService.isWithGroup(cvs, step) && !cvss.includes(cvs)) {
+            cvss.push(cvs);
```

Comment:
Probably it's fine in javascript but I don't like this pattern when we reference the `cvss` from outside of this function. I would prefer to pass it down as a parameter.